akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -272,9 +279,117 @@ object Source {
    *
    * @see [[Source.lazily]]
    */
+  @deprecated("Use 'Source.lazyCompletionStage' instead", "2.6.0")
   def lazilyAsync[T](create: function.Creator[CompletionStage[T]]): Source[T, Future[NotUsed]] =
     scaladsl.Source.lazilyAsync[T](() => create.create().toScala).asJava
 
+  /**
+   * Emits a single value when the given Scala `Future` is successfully completed and then completes the stream.
+   * The stream fails if the `Future` is completed with a failure.
+   *
+   * Here for Java interoperability, the normal use from Java should be [[Source.completionStage]]
+   */
+  def future[T](futureElement: Future[T]): Source[T, NotUsed] =
```

Comment:
In which situations do we provide also `Future` variants in the javadsl?

Reply:
Mostly for consistency because the Scala API has both future and completion stage. I don't have a strong argument for keeping it though.